cleverhans_tutorials/mnist_tutorial_cw.py

```diff
@@ -104,7 +107,7 @@ def mnist_tutorial_cw(train_start=0, train_end=60000, test_start=0,
     # Evaluate the accuracy of the MNIST model on legitimate test examples
     eval_params = {'batch_size': batch_size}
     accuracy = model_eval(sess, x, y, preds, X_test, Y_test, args=eval_params)
-    assert X_test.shape[0] == test_end - test_start, X_test.shape
+#    assert X_test.shape[0] == test_end - test_start, X_test.shape
```

Comment:
What's going on here?

If there's a good reason to remove the test, it's fine to remove.

If there's no reason to remove it, I'd prefer to keep it. A bit of defensive programming helps to prevent bugs and catch them quickly when they arise.

We generally don't want commented code in the repo.

Reply:
I commented this line as test_end and test_start are hard coded and depend on the dataset(number of training/test samples) hence making the code specific to MNIST. I am not sure for the purpose of having test_start and test_end. I felt it might be to control number of samples getting used but then its index specific too.

Reply:
The user can pass in any number for `test_start` and `test_end` so they're not specific to MNIST. The idea is that you can evaluate on just a subset of the test set if you want your evaluation to run fast. I think we use this in the tests.

Reply:
I am keeping the assert as of now. Can you please elaborate on the kind of bugs this assert is planned to detect? Will those be data loading bugs or model evaluation bugs?

I feel we should also keep an assert for test_end < len(test_data) immediately after the data loading.